src/domain/lattice.ml

```diff
@@ -651,3 +663,43 @@ struct
   let pretty_diff () ((x:t),(y:t)): Pretty.doc =
     Pretty.dprintf "%a not leq %a" pretty x pretty y
 end
+
+module Option (Base: S) : S with type t = Base.t option =
+struct
+  module Option = Printable.Option (Base) (struct let name = "None" end)
+  include Option
+
+  let binop f x y = match x, y with
+    | Some x, Some y -> Some (f x y)
+    | None, Some x
+    | Some x, None -> Some x
+    | None, None -> None
+
+  let leq x y = match x, y with
+    | Some x, Some y -> Base.leq x y
+    | Some x, None -> false
+    | None, _ -> true
```

Comment:
This seems to imply `None` is bot but then it really isn't? Is there any fix to get a proper lattice and still have it behave the way that is needed?